From d0f374b35141b4550538b78f6a8937fa88394cfb Mon Sep 17 00:00:00 2001
From: Eric Sun <jian.sun@freescale.com>
Date: Mon, 20 Aug 2012 20:51:25 +0800
Subject: [PATCH 1930/2463] ENGR00217687 [MX6SL_ARM2/EVK] Fix no perfmon
 directory

The problem is caused because the board init routine don't add the
corresponding device node. Problem resolved after add them

Signed-off-by: Eric Sun <jian.sun@freescale.com>
---
 arch/arm/mach-mx6/board-mx6sl_arm2.c |    4 ++++
 arch/arm/mach-mx6/board-mx6sl_evk.c  |    4 ++++
 2 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6sl_arm2.c b/arch/arm/mach-mx6/board-mx6sl_arm2.c
index 8f05702..1d76abb 100755
--- a/arch/arm/mach-mx6/board-mx6sl_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6sl_arm2.c
@@ -1254,6 +1254,10 @@ static void __init mx6_arm2_init(void)
 	imx6sl_add_dcp();
 	imx6sl_add_rngb();
 
+	imx6q_add_perfmon(0);
+	imx6q_add_perfmon(1);
+	imx6q_add_perfmon(2);
+
 	pm_power_off = mx6_snvs_poweroff;
 }
 
diff --git a/arch/arm/mach-mx6/board-mx6sl_evk.c b/arch/arm/mach-mx6/board-mx6sl_evk.c
index 62daad6..7ef077f 100644
--- a/arch/arm/mach-mx6/board-mx6sl_evk.c
+++ b/arch/arm/mach-mx6/board-mx6sl_evk.c
@@ -1266,6 +1266,10 @@ static void __init mx6_evk_init(void)
 	imx6sl_add_dcp();
 	imx6sl_add_rngb();
 
+	imx6q_add_perfmon(0);
+	imx6q_add_perfmon(1);
+	imx6q_add_perfmon(2);
+
 	pm_power_off = mx6_snvs_poweroff;
 }
 
-- 
1.7.7.4

